virttest/qemu_vm.py

```diff
@@ -1106,8 +1106,41 @@ def add_qxl(qxl_nr, base_addr=29):
                 qxl_str += " -device qxl,id=video%d,addr=0x%x" % (index, addr)
             return qxl_str
 
-        def add_vga(vga):
-            return " -vga %s" % vga
+        def add_vga(devices, vga):
+            """Add primary vga device."""
+            default_machine_re = re.compile(".*")
+            isapc_machine_re = re.compile("isapc")
+            pci_bus = {'aobject': 'pci.0'}
+            isa_bus = None  # so far there is no ISA bus
+            vga_dev_map = {
+                # name: [(machine regex, driver, parent bus), ...]
+                # the default variant must be the last one
+                "std": [(isapc_machine_re, "isa-vga", isa_bus),
+                        (default_machine_re, "VGA", pci_bus)],
+                "cirrus": [(isapc_machine_re, "isa-cirrus-vga", isa_bus),
+                           (default_machine_re, "cirrus-vga", pci_bus)],
+                "vmware": [(default_machine_re, "vmware-svga", pci_bus)],
+                "qxl": [(default_machine_re, "qxl-vga", pci_bus)],
+                "virtio": [(default_machine_re, "virtio-vga", pci_bus)],
```

Comment:
The mapping looks to be right approach,

@luckyh have you thought to have the mapping in cfg file ? do you think it doesn't makes sense or any drawback on it.

keeping it in config file can help to add future devices and its regex/pci_bus/isa_bus from configuration file. please share your thoughts.